tools/src/main/java/org/apache/kafka/trogdor/agent/WorkerManager.java

```diff
@@ -306,38 +336,63 @@ public Void apply(String errorString) {
             try {
                 worker.taskWorker.start(platform, worker.status, haltFuture);
             } catch (Exception e) {
+                log.info("{}: Worker {} start() exception", nodeName, worker.toString(), e);
                 stateChangeExecutor.submit(new HandleWorkerHalting(worker,
-                    "worker.start() exception: " + e.getMessage(), true));
+                    "worker.start() exception: " + Utils.stackTrace(e), true));
             }
             stateChangeExecutor.submit(new FinishCreatingWorker(worker));
+        } catch (ExecutionException e) {
+            if (e.getCause() instanceof RequestConflictException) {
+                log.info("{}: request conflict while creating worker {} for task {} with spec {}.",
+                    nodeName, workerId, taskId, spec);
+            } else {
+                log.info("{}: Error creating worker {} for task {} with spec {}",
+                    nodeName, workerId, taskId, spec, e);
+            }
+            throw e.getCause();
```

Comment:
what if `e.getCause()` is null? I understand we will see NullPointerException from throwing null vs. actually this being a cause of ExecutionException.

Reply:
ExecutionException always wraps another exception, though.  It's used by the futures stuff specifically to hold another exception.